hadoop-hdds/interface-server/src/main/proto/ScmServerDatanodeHeartbeatProtocol.proto

```diff
@@ -407,6 +409,20 @@ message ReplicateContainerCommandProto {
   required int64 cmdId = 3;
 }
 
+/**
+* This command asks the datanode to reconstruct the missing EC containers by
+* using remaining containers from sources.
+*/
+message ReconstructECContainersCommandProto {
+  required int64 containerID = 1;
+  repeated DatanodeDetailsProto sources = 2;
```

Comment:
I think it could be worth creating a simple type to pass datanodeDetails and replicaIndex together, eg:

```
message DatanodeDetailsAndReplicaIndexProto {
    required DatanodeDetailsProto datanodeDetails = 1;
    required int32 replicaIndex = 2;
}
```

Then have `repeated DatanodeDetailsAndReplicaIndexProto sources = 2`.

That way, we always send the two as a pair, and its less likely a mistake can be made serializing or de-serializing with order or one missing etc.

Reply:
Ah I got the answer for my above question. Let me create one structure now.